### ISSUE TITLE
Support multiple Docker Compose files

### ISSUE THREAD

**Issue by dieter-medium** [spring-projects/spring-boot#41691]
At the moment, the configuration option spring.docker.compose.file supports only a single file. However, I am using Spring Boot together with DevContainers, and to get my setup up and running, I need to use two Docker Compose files:

    ../compose.yaml
    ./compose.override.yaml

It would be highly beneficial if spring.docker.compose.file could support multiple files, similar to how Docker Compose itself allows specifying multiple files using the -f option.

**Benefits:**

- Aligns with Docker Compose's native functionality, offering a more flexible and powerful configuration.
- Supports complex setups like development with DevContainers.
- Improves user experience by reducing the need for workaround solutions.

**Additional Context:**

Supporting multiple Docker Compose files is a common scenario in development environments where configurations are split for better manageability. Implementing this feature would make Spring Boot's Docker Compose support more robust and versatile.

Thank you for considering this enhancement!

**Comment by mhalbritter** [comment]
Thanks for that proposal, I think this makes sense. How configuration merging works is documented here: https://docs.docker.com/compose/multiple-compose-files/merge/

**Comment by hannah23280** [comment]
Second to this
Currently, i wish i can specify multiple files, such that if the first one cannot be found, then go for the second one

```
  docker:
    compose:
      enabled: true
      lifecycle-management: start_only
      file:
        - a/docker-compose.yaml
        - b/docker-compose.yaml
   ```

**Comment by wilkinsona** [comment]
> Currently, i wish i can specify multiple files

That is exactly what this issue implemented. It has been available since Spring Boot 3.4

>  such that if the first one cannot be found, then go for the second one

That's a different requirement. If you would like us to consider support it, please open a new issue explaining why it would be useful.
